### PR TITLE
Support for Pebble Notices

### PR DESCRIPTION
**Pull request by benhoyt**
Just a WIP draft at the mojment.

See the spec: [JU048](https://docs.google.com/document/d/16PJ85fefalQd7JbWSxkRWn0Ye-Hs8S1yE99eW7pk8fA/edit).

### CODE REVIEW

File: internals/overlord/state/notices.go [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+			notice.lastRepeated = now
+			newOrRepeated = true
+		}
+	}
+	notice.lastOccurred = now
+	notice.lastData = data
+	notice.repeatAfter = repeatAfter
+
+	if newOrRepeated {
+		s.processNoticeWaiters()
```
Comment (pedronis) [inline internals/overlord/state/notices.go:212]: it feels a bit dangerous that this can potential block on unrelated code and timeouts decided by it, it would a bit more complex but safer to have in-between goroutines that distribute things to waiters

Comment (benhoyt) [reply internals/overlord/state/notices.go:212]: The idea here is that `processNoticeWaiters` can't block, because it only sends to the current `noticeWaiters` channels, and those are added and removed with each call to `WaitNotice` (the other goroutines). And we have a done (`ctx.Done()`) channel on each send too, to avoid blocking if the context is cancelled.

The `addNoticeWaiter` / `removeNoticeWaiter` / `processNoticeWaiters` logic is intentionally very close together in that file, partly for that reason.

So I'm not sure how adding additional in-between goroutines will help here?

**Comment by benhoyt** [comment]
Closing this draft in favour of the real PRs, starting with https://github.com/canonical/pebble/pull/295 (other PRs linked there).
